src/django/pfb_analysis/views.py

```diff
@@ -164,12 +164,17 @@ def get_serializer_class(self):
             return AnalysisLocalUploadTaskSerializer
 
     def perform_create(self, serializer):
+        if not serializer.is_valid():
+            return
         neighborhood_id = serializer.validated_data['neighborhood']
+        url = serializer.validated_data['upload_results_url']
         neighborhood = Neighborhood.objects.get(pk=neighborhood_id)
         user = self.request.user
         job = AnalysisJob.objects.create(neighborhood=neighborhood,
                                          created_by=user, modified_by=user)
-        serializer.save(job=job, created_by=user, modified_by=user)
+        serializer.save(job=job, upload_results_url=url, created_by=user, modified_by=user)
```

Comment:
It shouldn't be necessary to add `upload_results_url=url` here. By default, the serializer saves all properties in `validated_data`. The things passed here explicitly are passed because they're required model properties that require a value and aren't already part of `validated_data`.